configs/katello/nightly.yaml

```
@@ -33,3 +33,46 @@
       - centos-sclo-rh-rhel-7
       - centos-7-server-updates
       - centos-7-server
+  - name: katello-nightly-pulp3-rhel7
+    based_off: null
+    arches:
+      - x86_64
+    build_target: katello-nightly-pulp3-rhel7-build
+    external_repos:
+      - centos7-os
+      - centos7-updates
+    build_groups:
+      build:
+        - bash
+        - bzip2
+        - coreutils
+        - cpio
+        - diffutils
+        - findutils
+        - gawk
+        - gcc
+        - gcc-c++
+        - grep
+        - gzip
+        - info
+        - make
+        - patch
+        - python3-rpm-macros
+        - redhat-release-server
```

I don't think this exists in centos.
```suggestion
        - centos-release
```

---

Eh you're right.  It doesn't break anything, but it's also not correct either.

Although, i'd prefer "redhat-release" instead, as it's more general, and works across all redhat-family distros.  Will make the change when I revisit this PR

---

Don't think that works:
```console
$ yum info redhat-release
Loaded plugins: fastestmirror
Determining fastest mirrors
 * base: ams.edge.kernel.org
 * centos-sclo-rh: centos.mirror.triple-it.nl
 * epel: download.fedoraproject.org
 * extras: ams.edge.kernel.org
 * updates: centos.mirror.liteserver.nl
Error: No matching Packages to list
```

---

try `yum whatprovides redhat-release`

---

```
# yum whatprovides redhat-release
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
 * base: packages.oit.ncsu.edu
 * extras: packages.oit.ncsu.edu
 * updates: packages.oit.ncsu.edu
centos-release-7-7.1908.0.el7.centos.x86_64 : CentOS Linux release file
Repo        : base
Matched from:
Provides    : redhat-release = 7.7-10



centos-release-7-6.1810.2.el7.centos.x86_64 : CentOS Linux release file
Repo        : @CentOS
Matched from:
Provides    : redhat-release = 7.6-1
```

---

I'm not sure that's better. If the build was switched to another distro, it might provide more. On RHEL, isn't redhat-release different from redhat-server-release in what it pulls in?

---

@ekohl  So, every environment has only one package that provides "redhat-release", and this is to set up specific artifacts about the overall redhat-based release for the target environment.

In rhel proper, this is changed based on what flavor is installed.  You will only ever see one of "redhat-release-server", "redhat-release-workstation", "redhat-release-client", etc... They all provide "redhat-release", and they provide conflicting files.  I am pretty sure there isn't a "redhat-release" package itself, only other packages providing it. 

Here are the following files provided by centos-release:
```
centos-release-7-7.1908.0.el7.centos.x86_64/etc/centos-release
centos-release-7-7.1908.0.el7.centos.x86_64/etc/centos-release-upstream
centos-release-7-7.1908.0.el7.centos.x86_64/etc/issue
centos-release-7-7.1908.0.el7.centos.x86_64/etc/issue.net
centos-release-7-7.1908.0.el7.centos.x86_64/etc/os-release
centos-release-7-7.1908.0.el7.centos.x86_64/etc/pki/rpm-gpg
centos-release-7-7.1908.0.el7.centos.x86_64/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
centos-release-7-7.1908.0.el7.centos.x86_64/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Debug-7
centos-release-7-7.1908.0.el7.centos.x86_64/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Testing-7
centos-release-7-7.1908.0.el7.centos.x86_64/etc/redhat-release
centos-release-7-7.1908.0.el7.centos.x86_64/etc/rpm/macros.dist
centos-release-7-7.1908.0.el7.centos.x86_64/etc/system-release
centos-release-7-7.1908.0.el7.centos.x86_64/etc/system-release-cpe
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-Base.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-CR.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-Debuginfo.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-Media.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-Sources.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-Vault.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum.repos.d/CentOS-fasttrack.repo
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum/vars/contentdir
centos-release-7-7.1908.0.el7.centos.x86_64/etc/yum/vars/infra
centos-release-7-7.1908.0.el7.centos.x86_64/usr/lib/systemd/system-preset/85-display-manager.preset
centos-release-7-7.1908.0.el7.centos.x86_64/usr/lib/systemd/system-preset/90-default.preset
centos-release-7-7.1908.0.el7.centos.x86_64/usr/share/centos-release/EULA
centos-release-7-7.1908.0.el7.centos.x86_64/usr/share/doc/centos-release/Contributors
centos-release-7-7.1908.0.el7.centos.x86_64/usr/share/doc/centos-release/GPL
centos-release-7-7.1908.0.el7.centos.x86_64/usr/share/doc/redhat-release
centos-release-7-7.1908.0.el7.centos.x86_64/usr/share/redhat-release

```

What is relevant, in the case of our koji build system, is the following file:

```
centos-release-7-7.1908.0.el7.centos.x86_64/etc/rpm/macros.dist
```
and the following macros provided by the contents of the file:

```
# dist macros.

%centos_ver 7
%centos 7
%rhel 7
%dist .el7
%el7 1
```


For comparision, here is fedora-release-common:
```
fedora-release-common-31-1.noarch/etc/fedora-release
fedora-release-common-31-1.noarch/etc/issue
fedora-release-common-31-1.noarch/etc/issue.d
fedora-release-common-31-1.noarch/etc/issue.net
fedora-release-common-31-1.noarch/etc/os-release
fedora-release-common-31-1.noarch/etc/redhat-release
fedora-release-common-31-1.noarch/etc/swid
fedora-release-common-31-1.noarch/etc/swid/swidtags.d
fedora-release-common-31-1.noarch/etc/swid/swidtags.d/fedoraproject.org
fedora-release-common-31-1.noarch/etc/system-release
fedora-release-common-31-1.noarch/etc/system-release-cpe
fedora-release-common-31-1.noarch/usr/lib/fedora-release
fedora-release-common-31-1.noarch/usr/lib/issue
fedora-release-common-31-1.noarch/usr/lib/issue.net
fedora-release-common-31-1.noarch/usr/lib/rpm/macros.d/macros.dist
fedora-release-common-31-1.noarch/usr/lib/swidtag/fedoraproject.org
fedora-release-common-31-1.noarch/usr/lib/swidtag/fedoraproject.org/org.fedoraproject.Fedora-31.swidtag
fedora-release-common-31-1.noarch/usr/lib/system-release-cpe
fedora-release-common-31-1.noarch/usr/lib/systemd/system-preset
fedora-release-common-31-1.noarch/usr/lib/systemd/system-preset/85-display-manager.preset
fedora-release-common-31-1.noarch/usr/lib/systemd/system-preset/90-default.preset
fedora-release-common-31-1.noarch/usr/lib/systemd/system-preset/99-default-disable.preset
fedora-release-common-31-1.noarch/usr/lib/systemd/user-preset
fedora-release-common-31-1.noarch/usr/lib/systemd/user-preset/90-default-user.preset
fedora-release-common-31-1.noarch/usr/lib/variant
fedora-release-common-31-1.noarch/usr/share/licenses/fedora-release-common
fedora-release-common-31-1.noarch/usr/share/licenses/fedora-release-common/Fedora-Legal-README.txt
fedora-release-common-31-1.noarch/usr/share/licenses/fedora-release-common/LICENSE
```
Macros file:
```
fedora-release-common-31-1.noarch/usr/lib/rpm/macros.d/macros.dist
```
contents
```
# dist macros.

%fedora              31
%__bootstrap         ~bootstrap
%dist                %{!?distprefix0:%{?distprefix}}%{expand:%{lua:for i=0,9999 do print("%{?distprefix" .. i .."}") end}}.fc31%{?with_bootstrap:~bootstrap}
%fc31                1

``` 

This is quite literally the way we end up with the appropriate %dist macros on the target systems.  If we leave it at "redhat-release", and change out the underlying repos to different distros, (i.e. if we wanted fedora instead of centos), the correct thing will just be installed.

---

Thanks for explaining. It sounds like we should modify our various configs to use this and be consistent. Right now it's all over the place.